junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/discovery/DiscoverySelectorResolver.java

```diff
@@ -64,7 +64,7 @@ public void resolveSelectors(EngineDiscoveryRequest request, TestDescriptor engi
 		request.getSelectorsByType(UniqueIdSelector.class).forEach(selector -> {
 			javaElementsResolver.resolveUniqueId(selector.getUniqueId());
 		});
-		pruneTree(engineDescriptor);
+		engineDescriptor.pruneTree();
```

Comment:
When the line is deleted, three `DiscoverySelectorResolverTests` fail:

- `resolvingSelectorOfNonTestMethodResolvesNothing()`
- `resolvingUniqueIdWithWrongParamsResolvesNothing()`
- `resolvingUniqueIdOfNonTestMethodResolvesNothing()`

What was the purpose to write these? Who is depending on the same assumption?